### DISCUSSION TITLE
GIMX implementation

### DISCUSSION THREAD

**Discussion by 5a796d** [sekigon-gonnoc/Pico-PIO-USB#33] [Ideas]
it would be good for gimx adapter users, it would solve the problem of having to use a pc

someone did it via arduino + usb host shield but only one device works and does not continue development

**Comment by sanjay900** [comment]
You can definitely do this sort of thing - I've done it myself with Xbox 360 controllers. Using gimx as a base doesn't really help as much as you would think as it's firmware is written for LUFA and that's quite different to tinyusb, so you would essentially be porting and rewriting most of it. Eventually I plan on releasing a tool that does this sort of thing for you, but also let's you build controlers from scratch and then only use host for auth. One other problem you have is that a pc has drivers for pretty much every controller you want to use with gimx, but you would have to code that into your firmware and that is a bit more difficult. I do have some ideas for that too but it does make the problem a little more difficult.

**Reply by 5a796d** [reply]
I understand the difference in programming, having the controller drivers source is it possible to do something? (Logitech wheel driving force pro)
i know, host for auth assumes this is the case

**Reply by sanjay900** [reply]
Anything is possible, it just isn't an easy process. What are you trying to make your wheel work on?

**Reply by 5a796d** [reply]
I would like to do on ps4

**Reply by sanjay900** [reply]
It defo would be possible to look at the code and build something that does exactly what you want. for the authentication side you can look at the gimx firmware, and instead of sending auth packets via serial, you just send them directly to the controller via tinyusb. for actually parsing the data from the wheel itself, you would have to work out what the hid structure is for your wheel, and then transpose that over to whatever the ps4 is expecting. Gimx should know what the wheel is expecting, so you should be able to read it for that bit, but you would need to plug the wheel into your pc and grab its HID descriptor and work out what axis/buttons you want to map over
